src/Internal/Payment/State/InitialState.php

```diff
@@ -118,13 +131,21 @@ public function start_processing( PaymentRequest $request ) {
 		if ( null !== $duplicate_payment_result ) {
 			return $duplicate_payment_result;
 		}
+
+		$context = $this->get_context();
+		$this->minimum_amount_service->verify_amount(
+			(string) $context->get_currency(),
+			(int) $context->get_amount()
```

Comment:
This looked weird at first, and now I see that it is done in order to allow tests to pass. Let's remove this (unsafe) casting, and add simple mocks to the context in tests:

```php
// Make sure that the minimum amount verification is properly called.
$this->mock_context->expects( $this->once() )->method( 'get_amount' )->willReturn( 1 );
$this->mock_context->expects( $this->once() )->method( 'get_currency' )->willReturn( 'usd' );
```

If this causes an issue in the real world, we should be able to catch it and fix it. Otherwise we might have some silent issues.

Reply:
> This looked weird at first, and now I see that it is done in order to allow tests to pass. 

@RadoslavGeorgiev - My original intention was not to pass tests. Instead, that's because `$context->get_curreny()` and `$context->get_amount()` can return `null` based on their signatures. 

However, in the code flow of `InitialState`, that's NOT possible since their respective setters are always called in advance as a part of the `populate_context_from_order` call.

Given that, if both of them are null, then this call would be like the following, which is still acceptable: 

```php
		$this->minimum_amount_service->verify_amount('', 0)
```

WDYT?

Reply:
Agreed that this can be changed in the future.
Addressed here 68c40f439ba2b3591ffbb70e6a34a68a0d42a11a